src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,54 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd.MM.yyyy");
+        StringBuilder builderOfValidRange = new StringBuilder();
+
+        for (String dataItem : data) {
+            String[] parts = dataItem.split(" ");
+            try {
+                Date date = dateFormat.parse(parts[0]);
+                Date fromDate = dateFormat.parse(dateFrom);
+                Date toDate = dateFormat.parse(dateTo);
+                if (date.compareTo(fromDate) >= 0 && date.compareTo(toDate) <= 0) {
+                    builderOfValidRange.append(parts[1]).append(" ")
+                            .append(Integer.parseInt(parts[2]) * Integer.parseInt(parts[3]))
+                            .append(",");
+                }
+                if (date.compareTo(fromDate) == date.compareTo(toDate)) {
+                    builderOfValidRange.append(parts[1]).append(" ")
+                            .append(0).append(",");
+                }
+            } catch (ParseException e) {
+                throw new RuntimeException("data format error");
+            }
+        }
```

Comment:
1. move this to the next `for` loop in your code. 
2. no need to store data in ` builderOfValidRange`, remove it at all. do required actions with `names` and `data` array in one loop